### PR TITLE
Add support for generating JSON fields

### PR DESCRIPTION
**Pull request by anivanovic**
Easiest implementation to support generating JSON fields I could think of.
Developers still need to change schema but at least generation does not fail
and generated file compiles.

### CODE REVIEW

**Comment by anivanovic** [comment]
> @anivanovic I made a similar contribution, I addressed more cases with JSON field, would you like to talk in Discord in order to push it together?

Sure! Can you give me Discord link? I am also in Slack #ent channel. So we can open a thread there also.

**Comment by a8m** [comment]
Thanks for the support @zeevmoney, and thanks @anivanovic for the contribution.
Here's a link to the Discord server. Feel free to create a thread there - https://discord.gg/qZmPgTE6RX

I also suggest you use `go/parser.ParseExpr` to construct and validate the expression for configuring the JSON type. Please, feel free to mention me in Discord 😃

File: schemast/printer_test.go [hunk 1]
```diff
@@ -104,6 +104,19 @@ func TestPrintAddImport(t *testing.T) {
 	require.Equal(t, matches, 1)
 }
 
+func TestPrintStructOnSameLine(t *testing.T) {
```
Comment (anivanovic) [inline schemast/printer_test.go:107]: Using `parser.ParseExp` breaks this check. Generated struct is put in file as
```
struct{
}{}
```

Do not know if this is non issue and we can live with or we would like to have formatting like before. All fields on one line.

Comment (zeevmoney) [reply schemast/printer_test.go:107]: image

**Review by zeevmoney — APPROVED** [review]
great job!

**Review by a8m — APPROVED** [review]

**Comment by a8m** [comment]
Thanks, @anivanovic for your contribution! ❤️

**Pull request merged**